content/en/observability_pipelines/processors/remap_ocsf/custom_configuration.md

```diff
@@ -0,0 +1,369 @@
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+preprocess:
+- function: parse_csv
+  source: message
+  dest: .
+  skip_empty: true
+  - null
+  - receive_time
+  - serial
+  - type
+  - subtype
+  - null
+  - 'wildcard*'
+  - time_generated
+```
+
+{{% /tab %}}
+
+{{% tab "JSON example" %}}
+
+```json
+{
+  "preprocess": [
+    {
+      "function": "parse_csv",
+      "source": "message",
+      "dest": ".",
+      "skip_empty": true,
+      "columns": [
+        null,
+        "receive_time",
+        "serial",
+        "type",
+        "subtype",
+        null,
+        "wildcard*",
+        "time_generated"
+      ]
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+
+In the examples above, the preprocessor expects an array object with seven or more columns. Since the first and sixth columns are `null`, they are dropped. The other six columns are mapped to:
+- `receive_time`
+- `serial`
+- `type`
+- `subtype`
+- `time_generated`
+- `wildcard`, which is set to the remaining columns of the array.
+
+### Mapping
+
+Each mapping entry consists of a `dest` path naming the destination field in the OCSF event and either a `source` path that names a field in the source event or a `value` that contains a literal constant to insert at that destination.
+
+A `source` path starting with `OCSF.` names a field in the output OCSF event that has been previously assigned.
+
+A `default` value may be named which is then used if the source paths do not exist before any further processing is done. The mapping may optionally also name a `function` to and/or a `lookup` table as described below.
+
+{{< tabs >}}
+{{% tab "YAML example" %}}
+
+```yaml
+mapping:
+- dest: activity_name
+  value: Create
+- dest: actor.session.issuer
+  source: authenticationContext.issuer.id
+  default: Unknown
+```
+
+{{% /tab %}}
+{{% tab "JSON example" %}}
+
+```json
+{
+  "mapping": [
+    {
+      "dest": "activity_name",
+      "value": "Create"
+    },
+    {
+      "dest": "actor.session.issuer",
+      "source": "authenticationContext.issuer.id",
+      "default": "Unknown"
+    }
+  ]
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Implicit mappings
+
+All enumerated name or label fields identified in the OCSF schema are converted to their sibling `id` field. For example, the string field `severity` is automatically converted to the numeric field `severity_id` based on the standard enum value table defined in the OCSF schema. If no matching value is found in the lookup table, the `id` field is set to `99` to represent `Other`.
+
+If one of the listed `profiles` in the metadata section is `datetime`, the mapping
+automatically has all numeric timestamps identified in the OCSF schema converted into the sibling field `{DEST}_dt`. For example, the numeric `time` field is converted into `time_dt`, which contains a string representation of that timestamp. No additional work is required to support the `datetime` profile.
```

Comment:
Is there a link to the OCSF schema we can add here?

Reply:
OCSF is composed of multiple schemas, the choice of which is specified by the `version`, `class`, and `profiles` given in the `metadata` section. The complete schema can be browsed here: https://schema.ocsf.io/